PLFM-9199 - Metadata tasks manager & integration test

nickgros opened the pull request.

nickgros committed:
PLFM-9199

- Manager & unit tests
- Complete service, synapse client, & integration tests

nickgros committed:
PLFM-9199 - fix integration test

john-hill commented on services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/curation/CurationTaskManagerImpl.java:
```diff
@@ -0,0 +1,121 @@
+    }
+
+    @Override
+    public CurationTask updateCurationTask(UserInfo userInfo, CurationTask toUpdate) {
```
better to have transaction annotations.

john-hill commented on services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/curation/CurationTaskManagerImplUnitTest.java:
```diff
@@ -0,0 +1,324 @@
+
+    @ParameterizedTest
+    @EnumSource(CurationTaskPropertiesType.class)
+    public void testCreateCurationTask_Success(CurationTaskPropertiesType type) {
```
with

john-hill reviewed (approved):
Switch to long id, transaction, and _ to with.

nickgros committed:
PLFM-9199 - code review changes

- Add manager transaction annotations
- Change CurationTask taskId type to long
- Fix unit test names to adhere to conventions

nickgros committed:
PLFM-9199

nickgros committed:
PLFM-9199

- fix impl signature
